<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Gabriel J. Odom, PhD, ThD</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="odom_cv_20230712_files/libs/clipboard/clipboard.min.js"></script>
<script src="odom_cv_20230712_files/libs/quarto-html/quarto.js"></script>
<script src="odom_cv_20230712_files/libs/quarto-html/popper.min.js"></script>
<script src="odom_cv_20230712_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="odom_cv_20230712_files/libs/quarto-html/anchor.min.js"></script>
<link href="odom_cv_20230712_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="odom_cv_20230712_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="odom_cv_20230712_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="odom_cv_20230712_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="odom_cv_20230712_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="media/davewhipp-print.css">
</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Gabriel J. Odom, PhD, ThD</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p style="text-align: center;">
Data scientist. Statistician. Avid learner. Ordained priest. <a href="mailto:gabriel.j.odom@gmail.com" class="email">gabriel.j.odom@gmail.com</a> | <a href="https://github.com/gabrielodom/" target="_blank">https://github.com/gabrielodom</a>
</p>
<hr>
<section id="profile" class="level2">
<h2 class="anchored" data-anchor-id="profile">Profile</h2>
<p>I am an assistant professor of biostatistics at Florida International University’s <a href="https://stempel.fiu.edu/">Stempel College of Public Health</a>. I create open-source software packages and databases to facilitate public health research, and I produce and analyze computational statistics methods for genetics and genomics research. I enjoy mentoring my doctoral students, and I show them the simple joys of writing beautiful code. I want my passion for reproducible data science to spur my students to become better scientists and better people. Overall, I strive to become a better steward of my discipline.</p>
<section id="interests" class="level3">
<h3 class="anchored" data-anchor-id="interests">Interests</h3>
<p>R/Bioconductor Package Development, Endpoint Science, Statistical Genetics/Genomics, Reproducibility in Data Science, Public Health Surveillance and Informatics, High-Dimensional Statistics, Computational Statistics, Matrix Theory, Bayesian Statistics, Spatial/Time Series, Ethics in Public Health and Medical Informatics</p>
</section>
<section id="application-areas" class="level3">
<h3 class="anchored" data-anchor-id="application-areas">Application Areas</h3>
<p>Substance use disorders, public health data reporting, cancers, Alzheimer’s and neuro-degenerative diseases and related neurotoxicology</p>
</section>
</section>
<section id="employment" class="level2">
<h2 class="anchored" data-anchor-id="employment">Employment</h2>
<p><code>08/2019 -</code> <strong>Assistant Professor</strong>, Department of Biostatistics<br>
Stempel College of Public Health<br>
Florida International University, Miami, FL<br>
40 hours / week<br>
Salary: $106,102.04 / 9 months; $141,469.04 / 12 months</p>
<p><br></p>
<p><code>08/2017 - 08/2019</code> <strong>Postdoctoral Associate</strong>, Division of Biostatistics,<br>
Department of Public Health Sciences<br>
University of Miami’s Miller School of Medicine, Miami, FL<br>
40 hours / week<br>
Salary: $60,000 / year</p>
<p><br></p>
<p><code>08/2013 - 08/2017</code> <strong>Graduate Assistant</strong>, Department of Statistical Sciences,<br>
Baylor University, Waco, TX<br>
20 hours / week<br>
Salary: $24,000 / year</p>
</section>
<section id="research" class="level2">
<h2 class="anchored" data-anchor-id="research">Research</h2>
<ul>
<li><em>Towards a framework for algorithmic bias and fairness in predicting treatment outcome for opioid use disorder</em> (NIH AIM-AHEAD 1OT2OD032581-02-267). Co-PI. Contact PI: Laura Brandt, City College of New York.</li>
<li><em>Individual Level Predictive Modeling of Opioid Use Disorder Treatment Response using Aggregate CTN Datasets</em> (NIDA UG1DA013035-17). Data scientist. PI: Dan Feaster, University of Miami</li>
<li><em>FIU RCMI: Research Capacity Core</em> (NIMHD 2U54MD012393-06 8588). Data Scientist. PI: Zoran Bursac, Florida International University.</li>
<li><em>Addressing community-level influences of HIV and COVID-19 disparities among people with HIV</em> (NIMHD 2U54MD012393-06 8587). Lead Data Scientist. PI: Diana Sheehan, Florida International University.</li>
<li><em>A Software Tool to Standardize Outcomes and Measure Algorithmic Racial Bias in Opioid Clinical Trials</em> (NIMHD FIU-RCMI Pilot AWD000000009108). PI: Gabriel Odom</li>
<li><em>Building blood based DNA methylation signatures for AD that are reflective of CNS changes</em> (NIA 1RF1AG061127-01). Subcontract PI. PI: Lily Wang, University of Miami</li>
<li><em>New Statistical Strategies for Comprehensive Analysis of Epigenomewide Methylation Data</em>: NIH-funded post-doc research. PI: Lily Wang, University of Miami</li>
<li><em>iPGDAC, An Integrative Proteogenomic Data Analysis Center for CPTAC</em>: NIH-funded post-doc research. PI: Bing Zhang, Baylor College of Medicine; Site PI and Supervisor: Xi Steven Chen, University of Miami</li>
<li><em>Self-Correcting Energy-Efficient Water Reclamation Systems for Tailored Water Reuse at Decentralized Facilities</em>: NSF-funded PhD research. PI: Tzahi Cath, Colorado School of Mines; Co-PI and Supervisor: Amanda S. Hering, Baylor University</li>
</ul>
</section>
<section id="education" class="level2">
<h2 class="anchored" data-anchor-id="education">Education</h2>
<p><code>2023</code> <strong>Graduate Certificate, Innovations for Substance Use Disorders</strong></p>
<ul>
<li>Johns Hopkins University, Carey Business School</li>
<li>“Innovation Pitch Competition” Finalist</li>
</ul>
<p><code>2019</code> <strong>Postdoctoral Associate, Biostatistics</strong></p>
<ul>
<li>The University of Miami, Miller School of Medicine</li>
<li>Mentors: Xi Steven Chen and Lily Wang</li>
</ul>
<p><code>2017</code> <strong>PhD, Statistical Science</strong></p>
<ul>
<li>Baylor University</li>
<li>Dissertation: “Three Examples of Linear Dimension Reduction”</li>
<li>Advisors: Dean M. Young and Amanda S. Hering</li>
<li>Honors: Conyer’s Doctoral Honors Fellow; TeaCHE Capstone in Higher Education</li>
</ul>
<p><code>2014</code> <strong>ThD, Sacred Theology</strong></p>
<ul>
<li>International Miracle Institute</li>
<li>Dissertation: “An Exploration of Biblical Perspectives as Applied to Personal, Business, and National Economics”</li>
<li>Advisor: Christian M. E. Harfouche</li>
<li>Honors: Vested as Ordained Presbyter</li>
</ul>
<p><code>2011</code> <strong>BA, Economics; BS, Mathematics</strong></p>
<ul>
<li>The University of West Florida</li>
<li>Thesis: “An Economic Approach to University Campus Parking Shortages”</li>
<li>Advisor: Morris L. Marx</li>
<li>Honors: <em>Summa Cum Laude</em>, First in Major (Economics); Kugelman Honors <em>Techne</em> Scholar</li>
</ul>
<p><code>2011</code> <strong>MDiv, BTh; Theology</strong> (<em>conjoint</em>)</p>
<ul>
<li>International Miracle Institute</li>
<li>Honors: Licensed Member of the Clergy</li>
</ul>
</section>
<section id="manuscripts" class="level2">
<h2 class="anchored" data-anchor-id="manuscripts">Manuscripts</h2>
<section id="peer-reviewed" class="level3">
<h3 class="anchored" data-anchor-id="peer-reviewed"><em>Peer-reviewed</em></h3>
<ul>
<li>“Data cleaning and harmonization of clinical trial data: Medication-assisted treatment for opioid use disorder”. Balise RR, Hu MC, Calderon AR, <strong>Odom GJ</strong>, Brandt L, Luo SX, and Feaster DJ. <em>PLoS One</em>. 2024. <a href="https://doi.org/10.1371/journal.pone.0312695" target="_blank">https://doi.org/10.1371/journal.pone.0312695</a></li>
<li>“Empirically contrasting urine drug screening‐based opioid use disorder treatment outcome definitions”. Brandt L*, <strong>Odom GJ*</strong>, Hu MC, Castro C, Balise RR, and the CTN‐0094 Team. <em>Addiction</em>. 2024. <a href="https://doi.org/10.1111/add.16494" target="_blank">https://doi.org/10.1111/add.16494</a> (* signifies co-first-authorship)</li>
<li>“Individual-Level Risk Prediction of Return to Use During Opioid Use Disorder Treatment”. Luo SX, Feaster DJ, Liu Y, Balise RR, Hu MC, Bouzoubaa B, <strong>Odom GJ</strong>, Brandt L, Pan Y, Hser Y, VanVeldhuisen P, Castillo F, Calderon AR, Rotrosen J, Saxon AJ, Weiss RD, Wall M, and Nunes EV. <em>JAMA Psychiatry</em>. 2023. <a href="https://doi.org/10.1001/jamapsychiatry.2023.3596" target="_blank">https://doi.org/10.1001/jamapsychiatry.2023.3596</a></li>
<li>“Capturing drug use patterns at a glance: An <span class="math inline">\(n\)</span>-ary word sufficient statistic for repeated univariate categorical values”. <strong>Odom GJ*</strong>, Brandt L*, Castro C, Luo SX, Feaster DJ, Balise RR, the CTN-0094 Consortium Team. <em>Plos One</em>. 2023. <a href="https://doi.org/10.1371/journal.pone.0291248" target="_blank">https://doi.org/10.1371/journal.pone.0291248</a> (* signifies co-first-authorship)</li>
<li>“Risk of Experiencing an Overdose Event for Patients Undergoing Treatment With Medication for Opioid Use Disorder”. Brandt L, Hu MC, Liu Y, Castillo F, <strong>Odom GJ</strong>, Balise RR, Feaster DJ, Nunes EV, and Luo SX. <em>American Journal of Psychiatry</em>. 2023. <a href="https://doi.org/10.1176/appi.ajp.20220312" target="_blank">https://doi.org/10.1176/appi.ajp.20220312</a></li>
<li>“Specific polysubstance use patterns predict relapse among patients entering opioid use disorder treatment”. Pan Y, Feaster DJ, <strong>Odom GJ</strong>, Brandt L, Hu MC, Weiss RD, Rotrosen J, Saxon AJ, Luo SX, and Balise RR. <em>Drug and Alcohol Dependence Reports</em>. 2022. <a href="https://doi.org/10.1016/j.dadr.2022.100128" target="_blank">https://doi.org/10.1016/j.dadr.2022.100128</a></li>
<li>“pathwayMultiomics: An R package for efficient integrative analysis of multi-omics datasets with matched or un-matched samples”. <strong>Odom GJ</strong>, Colaprico A, Silva T, Chen X, and Wang L. <em>Frontiers in Genetics</em>. 2021. <a href="https://doi.org/10.3389/fgene.2021.783713" target="_blank">https://doi.org/10.3389/fgene.2021.783713</a></li>
<li>“pathwayPCA: an R/Bioconductor Package for Pathway Based Integrative Analysis of Multi‐Omics Data. <strong>Odom GJ</strong>, Ban Y, Colaprico A, Liu L<sup>†</sup>, Silva T, Sun X, Pico A, Zhang B, Wang L, and Chen X. <em>Proteomics</em>. 2020. <a href="https://doi.org/10.1002/pmic.201900409" target="_blank">https://doi.org/10.1002/pmic.201900409</a></li>
<li>“Moonlight: A tool to interpret pathways to discover cancer driver genes”. Colaprico A, Olsen C, Cava C, Bailey M, <strong>Odom GJ</strong>, Terkelsen T, Silva T, Olsen A, Cantini L, Zinovyev A, Barillot E, Bertoli G, Castiglioni I, Noushmehr H, Chen X, Papaleo E, and Bontempi G. <em>Nature Communications</em>. 2020. <a href="https://doi.org/10.1038/s41467-019-13803-0" target="_blank">https://doi.org/10.1038/s41467-019-13803-0</a></li>
<li>“An evaluation of supervised methods for identifying DMRs in Illumina methylation arrays.” Saurav M*, <strong>Odom GJ*</strong>, Gao Z, Chen X, and Wang L. <em>Briefings in Bioinformatics</em>. 2018. <a href="http://dx.doi.org/10.1093/bib/bby085" target="_blank">http://dx.doi.org/10.1093/bib/bby085</a>. (* signifies co-first-authorship)</li>
<li>“Multi-state multivariate statistical process control.” <strong>Odom GJ</strong>, Newhart KB, Cath TY, and Hering AS. <em>Applied Stochastic Models in Business and Industry</em>. 2018. <a href="https://doi.org/10.1002/asmb.2333" target="_blank">https://doi.org/10.1002/asmb.2333</a>.</li>
<li><br></li>
</ul>
</section>
<section id="with-mentees" class="level3">
<h3 class="anchored" data-anchor-id="with-mentees"><em>With mentees</em></h3>
<ul>
<li>“Using tree-based models to identify factors contributing to trait negative affect in adults”. Cañizares C<sup>†</sup>, Gómez-Maquet Y, Ferro E, Torres CA, Agudelo DM, and <strong>Odom GJ</strong>. <em>BMC Psychology</em>. 2025. <a href="https://doi.org/10.1186/s40359-024-02245-z" target="_blank">https://doi.org/10.1186/s40359-024-02245-z</a></li>
<li>“Mangrove Forests of Biscayne Bay, FL, USA may Act as Sinks for Plastic Debris”. Paduani M<sup>†</sup>, Ross M, and <strong>Odom GJ</strong>. <em>Wetlands</em>. 2024. <a href="https://doi.org/10.1007/s13157-024-01785-9" target="_blank">https://doi.org/10.1007/s13157-024-01785-9</a></li>
<li>“No evidence of accelerated epigenetic aging among black heroin users: A case vs control analysis”. Jones JD, Martinez S, Gonzalez I<sup>†</sup>, <strong>Odom GJ</strong>, and Comer SD. <em>Addiction Neuroscience</em>. 2023. <a href="https://doi.org/10.1016/j.addicn.2023.100096" target="_blank">https://doi.org/10.1016/j.addicn.2023.100096</a></li>
<li>“Evaluating the effect of acute diesel exhaust particle exposure on P-glycoprotein efflux transporter in the blood-brain barrier co-cultured with microglia”. Aquino GV<sup>†</sup>, Dabi A, <strong>Odom GJ</strong>, Lavado R, Nunn K, Thomas K, Schackmuth B, Shariff N, Jarajapu M, Pluto M, Miller SR, Eller L, Pressley J, Patel RR, Black J, and Bruce ED. <em>Current Research in Toxicology</em>. 2023. <a href="https://doi.org/10.1016/j.crtox.2023.100107" target="_blank">https://doi.org/10.1016/j.crtox.2023.100107</a></li>
<li>“The effect of changing the military’s sexual assault laws on law enforcement investigative findings in the US Army.” Carpenter ER, Gonzalez I<sup>†</sup>, Garcia SJ, and <strong>Odom GJ</strong>. <em>Law and Human Behavior</em>. 2022. <a href="https://doi.org/10.1037/lhb0000489" target="_blank">https://doi.org/10.1037/lhb0000489</a></li>
<li>“Lessons Learned From Miami-Dade County’s COVID-19 Epidemic: Making Surveillance Data Accessible for Policy Makers”. Williams R<sup>†</sup>, Bursac Z, Trepka M, and <strong>Odom GJ</strong>. <em>Journal of Public Health Management and Practice</em>. 2021. <a href="https://doi.org/10.1097/PHH.0000000000001364" target="_blank">https://doi.org/10.1097/PHH.0000000000001364</a></li>
<li>“Evaluating the endothelial-microglial interaction and comprehensive inflammatory marker profiles under acute exposure to ultrafine diesel exhaust particles <em>in vitro</em>”. Aquino G<sup>†</sup>, Dabi A, <strong>Odom GJ</strong>, Zhang F, and Bruce E. <em>Toxicology</em>. 2021. <a href="https://doi.org/10.1016/j.tox.2021.152748" target="_blank">https://doi.org/10.1016/j.tox.2021.152748</a></li>
<li>“coMethDMR: Accurate identification of co-methylated and differentially-methylated regions in epigenome-wide association studies with continuous phenotypes”. Gomez L<sup>†</sup>, <strong>Odom GJ</strong>, Liu L<sup>†</sup>, Gao Z, Chen X, and Wang L. <em>Nucleic Acids Research</em>. 2019. <a href="https://doi.org/10.1093/nar/gkz590" target="_blank">https://doi.org/10.1093/nar/gkz590</a>.</li>
<li>† <em>indicates academic mentee / advisee / graduate student</em></li>
</ul>
</section>
<section id="submittedin-revision" class="level3">
<h3 class="anchored" data-anchor-id="submittedin-revision"><em>Submitted/In Revision</em></h3>
<ul>
<li>“Endpoints as Algorithms: A Case Study of Bias in Opioid Use Disorder Treatment Outcomes”. Castro C, <strong>Odom GJ</strong>, Au L, and Brandt L. <em>Journal of Medical Ethics</em>.</li>
<li>“Engaging an Advisory Board in Discussions about the Ethical Relevance of Algorithmic Bias and Fairness”. Au L, Castro C, <strong>Odom GJ</strong>, and Brandt L. <em>npj Digital Medicine</em>.</li>
<li>“From Clinical Trials to Real-World Impact: Introducing a Computational Framework to Detect Endpoint Bias in Opioid Use Disorder Research”. <strong>Odom GJ*</strong>, Brandt L*, Marker A, Giorgi S, Jainarain G, Schwartz HA, Au L, Castro C, and the ENDPOINT Consortium. <em>Drug and Alcohol Review</em>. (* signifies co-first-authorship)</li>
<li>“The Sensitivity of Bayesian Kernel Machine Regression (BKMR) to Data Distribution: A Comprehensive Simulation Analysis”. Hasan KT<sup>†</sup>, <strong>Odom GJ</strong>, Bursac Z, and Ibrahimou B. <em>Journal of Statistical Computation and Simulation</em>.</li>
</ul>
</section>
</section>
<section id="software" class="level2">
<h2 class="anchored" data-anchor-id="software">Software</h2>
<ul>
<li><strong>Odom GJ</strong> and Balise RR. 2023. <em>public.ctn0094extra: Helper Files for the CTN-0094 Relational Database</em>. R package version 1.0.0 version controlled with Git on GitHub, <a href="https://github.com/CTN-0094/public.ctn0094extra" target="_blank">https://github.com/CTN-0094/public.ctn0094extra</a>.</li>
<li>Balise RR, <strong>Odom GJ</strong>, Calderon A. 2023. <em>public.ctn0094data: De-Identified Data from CTN-0094</em>. R/CRAN package version 1.0.6, <a href="https://CRAN.R-project.org/package=public.ctn0094data" target="_blank">https://CRAN.R-project.org/package=public.ctn0094data</a>.</li>
<li>Cañizares C<sup>†</sup>, Macgowan MJ, and <strong>Odom GJ</strong>. 2022. <em>tidyDisasters: Disaster Data Sets Including FEMA, EMDAT and GTD Information</em>. R/CRAN package version 0.1.1, <a href="https://CRAN.R-project.org/package=tidyDisasters" target="_blank">https://CRAN.R-project.org/package=tidyDisasters</a>.</li>
<li><strong>Odom GJ</strong>, Brandt L, and Balise RR. 2022. <em>CTNote: Clinical Trials Network Outcomes, Treatments, and Endpoints</em>. R/CRAN package version 0.1.0, <a href="https://CRAN.R-project.org/package=CTNote" target="_blank">https://CRAN.R-project.org/package=CTNote</a>.</li>
<li>Rodriguez A<sup>†</sup>, Bursac Z, Trepka M, Williams R<sup>†</sup>, and <strong>Odom GJ</strong>. 2022. <em>COVID-19 Information for South Florida: an Interactive Shiny Dashboard</em>. <a href="https://annyrodriguez.shinyapps.io/COVID19/" target="_blank">https://annyrodriguez.shinyapps.io/COVID19/</a>.</li>
<li>Gomez L<sup>†</sup>, Veitzman F<sup>†</sup>, Silva T, Wang L, and <strong>Odom GJ</strong>. 2019. <em>coMethDMR: Apply DMR-detection methods to detect co-methylated regions of the genome</em>. R/Bioconductor package version 1.2.0, <a href="https://doi.org/10.18129/B9.bioc.coMethDMR" target="_blank">https://doi.org/10.18129/B9.bioc.coMethDMR</a>.</li>
<li>Balise R, Bouzoubaa L<sup>†</sup>, <strong>Odom GJ</strong>, and Castor N. 2021. <em>DOPE: a Drug Ontology Parsing Engine</em>. R/CRAN package version 2.1.0, <a href="https://CRAN.R-project.org/package=DOPE" target="_blank">https://CRAN.R-project.org/package=DOPE</a>.</li>
<li>Balise R and <strong>Odom GJ</strong>. 2020. <em>tidyREDCap: Helper functions for working with REDCap data</em>. R/CRAN package version 1.0.1, <a href="https://CRAN.R-project.org/package=tidyREDCap" target="_blank">https://CRAN.R-project.org/package=tidyREDCap</a>.</li>
<li>Balise R, <strong>Odom GJ</strong>, Grealis K, and Cardozo F. 2020. <em>rUM: R Templates from the University of Miami</em>. R/CRAN package version 1.0.1, <a href="https://CRAN.R-project.org/package=rUM" target="_blank">https://CRAN.R-project.org/package=rUM</a>.</li>
<li>Zhang L<sup>†</sup>, <strong>Odom GJ</strong>, Silva T, Gomez L<sup>†</sup>, and Wang L. 2020. <em>rnaEditr: Statistical analysis of RNA editing sites and hyper-editing regions</em>. R/Bioconductor package version 1.4.0, <a href="https://doi.org/10.18129/B9.bioc.rnaEditr" target="_blank">https://doi.org/10.18129/B9.bioc.rnaEditr</a>.</li>
<li><strong>Odom GJ</strong>, Ban J, Liu L<sup>†</sup>, Wang L, and Chen S. 2019. <em>pathwayPCA: Integrative pathway analysis with modern PCA methodology and gene selection</em>. R/Bioconductor package version 1.10.0, <a href="https://doi.org/10.18129/B9.bioc.pathwayPCA" target="_blank">https://doi.org/10.18129/B9.bioc.pathwayPCA</a>.</li>
<li><strong>Odom GJ</strong>, Barnard B, Johnson M, Kazor K, and Hering A. 2017. <em>mvMonitoring: Multi-state adaptive dynamic principal component analysis for multivariate process monitoring</em>. R/CRAN package version 0.1.0, <a href="https://CRAN.R-project.org/package=mvMonitoring" target="_blank">https://CRAN.R-project.org/package=mvMonitoring</a>.</li>
<li><br></li>
<li>† <em>indicates academic mentee / advisee / graduate student</em></li>
</ul>
</section>
<section id="patents" class="level2">
<h2 class="anchored" data-anchor-id="patents">Patents</h2>
<ul>
<li>“A DYNAMIC MULTI-FACTOR REPRESENTATION OF HEALTH DATA.” Kobetz E, Balise RR, Bailey Z, Dominguez S, Bouzoubaa L, Abranches G, Picado Roque O, Stoler J, Ewing C, and <strong>Odom GJ</strong>. 2021. Patent No.&nbsp;US20210383932A1. <a href="https://patentscope.wipo.int/search/en/detail.jsf?docId=WO2020086905" class="uri">https://patentscope.wipo.int/search/en/detail.jsf?docId=WO2020086905</a></li>
</ul>
<!-- Our Reference:  126683-003PV1. Your Reference: UMIP-193. -->
</section>
<section id="teaching" class="level2">
<h2 class="anchored" data-anchor-id="teaching">Teaching</h2>
<ul>
<li><em>Introduction to Data Science for Health Disparities</em>. Co-instructor. Morehouse School of Medicine. 2025.</li>
<li><em>Advanced Biostatistics III</em>. Instructor of Record, Florida International University’s Stempel College of Public Health. 2024.</li>
<li><em>Introduction to Bayesian Inference</em>. Instructor of Record, Florida International University’s Stempel College of Public Health. 2022-2023.</li>
<li><em>Advanced Biostatistics III</em>. Co-instructor, Florida International University’s Stempel College of Public Health. 2022.</li>
<li><em>Introduction to Statistical Methods in R</em>. Instructor of Record, Florida International University’s Stempel College of Public Health. 2021-2023.</li>
<li><em>Advanced R Computing for Public Health Sciences</em>. Instructor of Record, Florida International University’s Stempel College of Public Health. 2020-2023.</li>
<li><em>Multivariate Methods in Health Sciences Research</em>. Instructor of Record, Florida International University’s Stempel College of Public Health. 2020.</li>
<li><em>Survey of Statistical Computing</em>. Instructor of Record, University of Miami’s Miller School of Medicine. 2019.</li>
<li><em>Topics in Biostatistical Research I &amp; II</em>. Instructor of Record, University of Miami’s Miller School of Medicine. 2018 – 2019.
<ul>
<li><em>PhD sections</em>: coordinated topics with graduate and external faculty; developed course syllabi; wrote and administered all projects and presentations; assigned all grades</li>
<li><em>MS sections</em>: coordinated topics with graduate faculty; developed course syllabi and course notes; wrote and administered all homework assignments and projects; assigned all grades</li>
</ul></li>
<li><em>Complex Data Analysis</em>. Lab Instructor, University of Miami’s Miller School of Medicine. 2018-2019. Taught weekly software lab sessions to doctoral students in biostatistics and engineering</li>
<li><em>Quantitative Methods for Business I &amp; II</em>. Instructor of Record, Baylor University’s Hankamer School of Business. 2014 – 2016. Developed course syllabi, course notes, and slides; wrote and administered all quizzes, projects, homework assignments, and exams; assigned all grades</li>
<li><em>Introduction to Christian Scriptures</em>; <em>On the Sacraments</em>; <em>Topics in Leadership</em>. Instructor of Practice, Family Worship Center’s Leadership Academy, Waco, TX. 2014 – 2017. Developed course curriculum and projects; reviewed capstone projects; mentored new deacons</li>
</ul>
<section id="licenses" class="level3">
<h3 class="anchored" data-anchor-id="licenses">Licenses</h3>
<ul>
<li><em>Software Carpentry</em>. Licensed Instructor, 2018–present</li>
<li><em>Data Carpentry</em>. Licensed Instructor, 2018–present</li>
</ul>
</section>
<section id="doctoral-students" class="level3">
<h3 class="anchored" data-anchor-id="doctoral-students">Doctoral Students</h3>
<ul>
<li>Ingrid Gonzalez, biostatistics. <a href="https://stempel.fiu.edu/faculty-staff/profiles/gonzalez-ingrid.html" class="uri">https://stempel.fiu.edu/faculty-staff/profiles/gonzalez-ingrid.html</a></li>
<li>Tanvir Hasan, biostatistics. <a href="https://www.linkedin.com/in/kazitanvirhasan/" class="uri">https://www.linkedin.com/in/kazitanvirhasan/</a></li>
<li>Roy Williams, biostatistics. <a href="https://rwilli56.wixsite.com/website" class="uri">https://rwilli56.wixsite.com/website</a></li>
<li>Sheblie Burchfield, biostatistics. <a href="https://www.linkedin.com/in/shelbie-burchfield-ms-mph-01404a9a/" class="uri">https://www.linkedin.com/in/shelbie-burchfield-ms-mph-01404a9a/</a></li>
</ul>
</section>
</section>
<section id="presentations" class="level2">
<h2 class="anchored" data-anchor-id="presentations">Presentations</h2>
<ul>
<li>“Evaluating the Safety and Efficacy of Clinical Decision Support Algorithms”. Odom GJ and Brandt L. 2023. Johns Hopkins University’s Innovations for Substance Use Disorder Pitch Competition, <strong>Invited</strong> presentation. Baltimore, MD.</li>
<li>“A Gentle Introduction to Data Science Tools for Addiction Research: Use Cases With Examples”. Odom GJ and Balise RR. 2023. College on Problems of Drug Dependence. Denver, CO.</li>
<li>“R for Reproducible Scientific Analysis: A ‘Train the Trainers’ Workshop Series”. Odom GJ and Idris M. 2023. Research Centers in Minority Institutions Coordinating Center National Conference, <strong>Invited</strong> presentation. Bethesda, MD.</li>
<li>“Improving outcome of medication treatment for OUD: Comparing and validating treatment endpoints”. Odom GJ. 2022. American Academy of Addiction Psychiatry’s Annual Meeting and Scientific Symposium. Naples, FL.</li>
<li>“Introduction to Quarto”. Odom GJ. 2022. <strong>Invited</strong> workshop for the ASA’s Section of Statistical Programmers and Analysts. Virtual.</li>
<li>“Integrate Partially-Matched Multi-Omics Data with the MiniMax Statistic”. Odom GJ, Colaprico A, Silva S, Chen S, and Wang L. 2022. Joint Statistical Meetings of the ASA. Washington, DC.</li>
<li>“Empirically Comparing Opioid Use Disorder Treatment Outcomes”. Odom GJ, Brandt L, Balise RR. 2022. College on Problems of Drug Dependence. Minneapolis, MN.</li>
<li>“Detecting Regions of Concurrent and Differential Methylation with coMethDMR (R/Bioconductor)”. Odom GJ, Martinez M, Gonzalez I, and Jones J. 2022. NIDA Genetics and Epigenetics Cross-Cutting Meeting. Virtual.</li>
<li>“Identifying Differentially Methylated Genomic Regions with coMethDMR”. Odom GJ, Martinez M, Silva S, Gonzalez I, Veitzman F, Jones J. and Wang L. 2022. BioC2022. <a href="https://bioc2022.bioconductor.org/abstracts/paper10/" class="uri">https://bioc2022.bioconductor.org/abstracts/paper10/</a>. Seattle, WA.</li>
<li>“Using the MiniMax Statistic to Integrate Partially-Matched Multi-Omics Data”. Odom GJ and Wang L. 2022. FIU’s Biomolecular Sciences Institute lecture series. Miami, FL.</li>
<li>“Making Public Health Data Accesible to Local Policy Makers: Lessons Learned during the COVID-19 Pandemic”. Odom GJ, Williams R, Rodriguez A, Bursac Z, and Trepka M. 2021. FIU-RCMI’s Health Equity Symposium. Miami, FL</li>
<li>“Using the R tidyREDCap Package to Help Novices Explore Research Data”. Balise RR and Odom GJ. 2021. Latin American &amp; Brazilian REDCap Conference – Virtual; <strong>International Invited session</strong>.</li>
<li>“COVID-19 Data Visualization in Practice”. 2021. ASA’s Joint Statistical Meetings – Virtual.</li>
<li>“DOPE: A Package for the R Language to Process/Classify Drug Names”. Balise RR, Odom GJ, Bouzoubaa L, Luo S, and Feaster D. 2021. College on Problems of Drug Dependence – Virtual.</li>
<li>“Data Visualization and the R tidyREDCap Package”. Balise RR, Espinosa V, and Odom GJ. 2020. REDCapCon – Virtual; <strong>International Invited session</strong>.</li>
<li>“Integrative Pathway Analysis with pathwayPCA”. 2019. BioC 2019 Software Workshop – New York City, NY; <strong>Young Researcher Travel Award</strong></li>
<li>“Moonlight: A Tool to Interpret Pathways to Discover Cancer-driver Genes”. 2019. European Conference on Computational Biology – Basel, Switzerland; <em>poster presented by Antonio Colaprico</em></li>
<li>“Reproducible Data Science with R Packages”. 2019. <strong>Invited</strong> Lecture at the University of Tennessee College of Medicine – Memphis, TN</li>
<li>“Conducting Reproducible Research using R Packages”. 2019. <strong>Invited</strong> Lecture at Florida International University – Miami, FL</li>
<li>“R for Collaborative and Reproducible Research: Creating your First R Package”. 2018. Biostatistics and Bioinformatics Shared Resources <strong>Invited</strong> Lecture at the Sylvester Comprehensive Cancer Center – Miami, FL</li>
<li>“R for Collaborative and Reproducible Research: Project Organization and Style”. 2018. Biostatistics and Bioinformatics Shared Resources <strong>Invited</strong> Lecture at the Sylvester Comprehensive Cancer Center – Miami, FL</li>
<li>“R for Collaborative and Reproducible Research: Version Control with Git and GitHub”. 2018. Biostatistics and Bioinformatics Shared Resources <strong>Invited</strong> Lecture at the Sylvester Comprehensive Cancer Center – Miami, FL</li>
<li>“Early Fault Detection of Activated Sludge Treatment Systems Utilizing Multivariate Statistical Analysis”. 2018. WEFTEC Student Poster Sessions – New Orleans, LA; <em>poster presented by Kathryn Newhart</em></li>
<li>“Data Visualization with the <code>ggplot</code> and <code>ggmap</code> Packages”. 2018. Quantitative Science Clinic <strong>Invited</strong> Lecture at the Sylvester Comprehensive Cancer Center – Miami, FL</li>
<li>“Data Analysis Workflow in R using the RStudio Integrated Development Environment”. 2018. Quantitative Science Clinic <strong>Invited</strong> Lecture at the Sylvester Comprehensive Cancer Center – Miami, FL</li>
<li>“Data Manipulation and Transformation with the <code>dplyr</code> Package”. 2018. Quantitative Science Clinic <strong>Invited</strong> Lecture at the Sylvester Comprehensive Cancer Center – Miami, FL</li>
<li>“Exploratory Data Analysis Walkthrough with the <code>tidyverse</code> Package Suite”. 2018. Quantitative Science Clinic <strong>Invited</strong> Lecture at the Sylvester Comprehensive Cancer Center – Miami, FL</li>
<li>“Multi-State Multivariate Statistical Process Control”. 2017. Quality and Productivity Research Conference <strong>Invited</strong> Session – Storrs, CT</li>
<li>“Iterated Linear Dimension Reduction with Application to Misclassification Reduction”. 2016. Southern Regional Council on Statistics Student Poster Session – Bentonville, AR; <strong>Outstanding Student Poster Award</strong></li>
<li>“Two-Step LDR for Conditional Error Rate Reduction”. 2016 Conference of Texas Statisticians Student Poster Session – San Antonio, TX</li>
<li>“On University Campus Parking Shortages and Possible Solutions”. 2011. Southern Regional Honors Council – Little Rock, AR; Honors Thesis Defense</li>
</ul>
</section>
<section id="honors" class="level2">
<h2 class="anchored" data-anchor-id="honors">Honors</h2>
<p><code>2023</code> Doctorate of Ministry, <u>honoris causa</u>. International Miracle Institute.</p>
<p><code>2022</code> Certificate of Merit, the City of Miami Office of the Mayor. For service to the city during the COVID-19 pandemic.</p>
<p><code>2021</code> FIU Presidential Service Award (<em>runner up</em>) for service to the university community via the Miami COVID Project (Mary Jo Trepka, Zoran Bursac, Gabriel J. Odom, and Roy Williams)</p>
<p><code>2020</code> <strong>Sigma Xi Scientific Research Honor Society</strong>, Florida International University</p>
<p><code>2019</code> Voluntary Assistant Professor, The University of Miami Department of Public Health Sciences</p>
<p><code>2019</code> Young Researcher Travel Award, Bioconductor’s <em>BioC 2019</em> Conference</p>
<p><code>2018</code> <strong>Lifetime Achievement Award</strong> for Community Service, <em><strong>The President of the United States</strong></em></p>
<p><code>2017</code> <strong>Phi Beta Kappa Society</strong>, Baylor University</p>
<p><code>2016</code> R. L. Anderson Student Presentation Award, Southern Regional Council on Statistics</p>
<p><code>2016</code> Outstanding Alumni Award (nominated), University of West Florida</p>
<p><code>2016</code> Phi Kappa Chi Outstanding Instructor Award, Baylor University</p>
<p><code>2015 - 2016</code> Department of Statistical Science Ambassador Award, Baylor University</p>
<p><code>2014 – 2016</code> Outstanding Graduate Instructor (finalist), Baylor University</p>
<p><code>2013 – 2017</code> Dean’s Fellowship, Baylor University</p>
<p><code>2014</code> <strong>Golden Key International Honours Society</strong>, Baylor University</p>
<p><code>2014</code> <strong>Omicron Delta Kappa National Leadership Honors Society</strong>, Baylor University</p>
<p><code>2013</code> The Lion Award for Workplace Excellence, North Central Texas College</p>
<p><code>2011</code> <strong>Phi Kappa Phi Honors Society</strong>, University of West Florida</p>
<p><code>2011</code> Pi Mu Epsilon Honors Society of Mathematics, University of West Florida</p>
<p><code>2011</code> Scholastic Excellence Award, University of West Florida</p>
<p><code>2010</code> Honors <strong>Mentor of the Year</strong>, University of West Florida</p>
<p><code>2008 – 2011</code> President’s List, University of West Florida</p>
<p><code>2008</code> Phi Theta Kappa Honors Society</p>
</section>
<section id="service" class="level2">
<h2 class="anchored" data-anchor-id="service">Service</h2>
<ul>
<li><em>Doctoral Progam Comprehensive Examination Committee</em>, FIU’s Department of Biostatistics. 2023-2025. Wrote and graded PhD comprehensive exams.</li>
<li><em>Doctoral Progam Admittance Committee</em>, FIU’s Department of Biostatistics. 2021-2025. Reviewed applicant vitae, interviewed candidates, and nominated top candidates for university graduate scholarships.</li>
<li><em>Grant Reviewer</em>, NIDA. 2022. Served on the grant review panel for the “Accelerating the Pace of Drug Abuse Research Using Existing Data” study section.</li>
<li><em>Doctoral Program Development Committee</em>, FIU’s Department of Biostatistics. 2019 - 2021. Wrote the new degree application, syllabi for new courses, and chaired the hiring committee for new graduate faculty in the program.</li>
<li><em>Lead Data Scientist</em>, The Miami COVID Project. 2020 - 2021. Cleaned and wrangled publich health data related to the COVID-19 epidemic in Miami-Dade, Broward, and Palm Beach counties in Florida; gave weekly presentations on the local state of the pandemic and trends to county and city elected officials, presented our findings to the general public via a weekly-updated website: <a href="https://rwilli5.github.io/MiamiCovidProject/" target="_blank">https://rwilli5.github.io/MiamiCovidProject/</a></li>
<li><em>Program Committee Chair Elect and Chair</em>, American Statistical Association’s Section of Statistical Programmers and Analysts. 2020 - 2022. Assisted in the curation of the sections invited talks for the ASA’s Joint Statistical Meetings in 2022</li>
<li><em>PhD Program Development Committee</em>, Department of Biostatistics, Florida International University. 2019 - 2021. Developed new curriculum schedule for the PhD in biostatistics, co-wrote the application module presented to the State of Florida Board of Governors, and wrote, modified, and/or extensively edited all submitted syllabi</li>
<li><em>Seminar Series Coordinator</em>, Division of Biostatistics, University of Miami. 2018 – 2019. Curated list of nationally-recognized speakers in biostatistics; invited these speakers to the Miller School of Medicine for grand rounds; coordinated and hosted guest faculty during their visit</li>
<li><em>Assistant Clergy in Residence</em>, Words of Life Fellowship Church, Miami, FL. 2017 – 2019. Mentored parishioners in service and ministry roles; performed clerical visits</li>
<li><em>Assistant Clergy in Residence</em>, Family Worship Center, Waco, TX. 2014 – 2017. Wrote sermons and homilies; mentored deacons in service and ministry roles; performed clerical visits; taught leadership academy candidates</li>
</ul>
<p><br></p>
<hr>
<p><br></p>
<!--
### Footer
Last updated: November 2023
Last updated: September 2023
Last updated: July 2023
Last updated: March 2023
Last updated: November 2022
Last updated: February 2022
Last updated: November 2021
Last updated: February 2020
-->
<p><em>Updated: 2025-03-03</em></p>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>